website/project/decorators.py

```diff
@@ -84,6 +84,8 @@ def must_be_valid_project_inner(func):
         @functools.wraps(func)
         def wrapped(*args, **kwargs):
             if preprints_valid and Preprint.load(kwargs.get('pid')):
+                _inject_nodes(kwargs)
```

Comment:
`must_be_valid_project` was modified as a part of the divorce if `preprints_valid=True` passed in, then it would work for preprints.  called `_inject_nodes` will add kwargs['node'] = Preprint object.  Then `claim_user_registered` will have the Preprint object as `node`, and all methods in that function will work for a node or preprint.

There might be some confusion over "node" language referring to a preprint - at some point, we might want to generalize the language in a few places that nodes and preprints both use, (resource, or similar), but I think this is low priority.